horovod/torch/__init__.py

```diff
@@ -157,13 +159,31 @@ def synchronize(self):
 
         self._synchronized = True
 
-    def step(self, closure=None, synchronize=True):
-        if synchronize:
+    @contextmanager
+    def already_synchronized(self):
+        """
+        A context manager used to specify that optimizer.step() should
+        not perform synchronization.
+
+        It's typically used in a following pattern:
```

Comment:
nit: It's typically used in a following pattern: >> It's typically used in the following pattern: